CMakeLists.txt

```diff
@@ -148,10 +148,29 @@ if (USE_TLS)
     endif()
 endif()
 
-add_library( ixwebsocket
-    ${IXWEBSOCKET_SOURCES}
-    ${IXWEBSOCKET_HEADERS}
-)
+if(BUILD_SHARED_LIBS)
+    # Building shared library
+    
+    if(MSVC)
+        # Workaround for some projects
+        set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+    endif()
+
+    add_library( ixwebsocket SHARED
+        ${IXWEBSOCKET_SOURCES}
+        ${IXWEBSOCKET_HEADERS}
+    )
+    
```

Comment:
That part will be annoying, we could try to read it from the ixwebsocket/IXVersion.h (forgot the name) instead.

Reply:
It’s better then to move version definition to cmake file and then use female constants in the source files to avoid hard-coded definitions. It’s impractical to parse include file when it could be done in opposite way.